### ISSUE TITLE
The X11 connection broke: I/O error (code 1)

### ISSUE THREAD

**Issue by jacknlliu** [osrf/docker_images#40]
I got the following error with  `osrf/ros:kinetic-desktop-full` docker image in Fedora 24.
```
The X11 connection broke: I/O error (code 1)
XIO:  fatal IO error 2 (No such file or directory) on X server ":0"
      after 445 requests (439 known processed) with 0 events remaining.
```

The gui program cannot work in this docker image.
Any help will be appreciated!

**Comment by ruffsl** [comment]
Just tested the latest image on ubuntu 16.04 (I don't have a Fedora install), and GUI's using X11 are working for me via the simple method way show here: http://wiki.ros.org/docker/Tutorials/GUI

Is your default display number different than `:0` in Fedora, is Fedora still using X server, not Wayland?
https://ask.fedoraproject.org/en/question/90679/cannot-start-x-server-on-f24/

**Comment by jacknlliu** [comment]
The `osrf/ros:kinetic-desktop-full` image works well on Ubuntu 16.04, but not the Fedora 24. The Fedora 24 still using X server and my display number is `:0`. 

But the `osrf/ros:indigo-desktop-full` images works well on Fedora 24.

Is this the limitation of docker ? Something limit the docker app running on any Linux distribution?

**Comment by ruffsl** [comment]
Not sure how this would be docker specific issue if it works for one docker image but not another. I'd suspect it to be more of a Ubuntu:Xenial/Fedora-24 image/host related issue as Ubuntu:Trusty/Fedora-24 seem to be working. 

Could you verify this is the case by checking if say: the gedit GUI works from Xenial docker image running on a Fedora 24 host?

``` dockerfile
FROM ubuntu:16.04
RUN apt-get update && apt-get install -y gedit
```

**Comment by jacknlliu** [comment]
I got the following error run `gedit`  with ubuntu:16.04 docker image on Fedora 24 host
```
Failed to connect to Mir: Failed to connect to server socket: No such file or directory
Unable to init server: Could not connect: Connection refused

(gedit:13): Gtk-WARNING **: cannot open display: :0
```

The `gedit` cannot work.

I used the following command to run the docker image,
```
docker   run   --device   /dev/dri --env="DISPLAY" --env QT_X11_NO_MITSHM=1  --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" -it    ubuntu/test
```

**Comment by ruffsl** [comment]
Hmmm, not sure where Mir is coming into this. Are new images of Ubuntu 16:04 now using Mir by default? 
I had thought that display transition would've been made after this LTS.

**Comment by jacknlliu** [comment]
Yeah, the Ubuntu 16.04 use Mir as default display server, so the X applications in docker container can't find the Mir server in Fedora which still use X server.  

This is the drawback of docker to run GUI application in docker container. At this aspect,  docker container is not better than virtual machine.

**Comment by ruffsl** [comment]
Well, that's the cost of adopting different display technologies; the diversity pushes innovation and security at the cost of convenience and uniformity. Still, I kind of wish Canonical had adopted Wayland instead. 
A good discussion on the topic from 6 months ago:
https://www.reddit.com/r/linux/comments/4bq9kl/eli5_wayland_vs_mir_vs_x11/

Perhaps you could follow online guides to install Wayland packages onto the Ubuntu image, and then mount the necessary components for Wayland to forward the GUI in the container to the host display. No one has yet contributed to that section of the ros docker wiki, and I'd hazard a guess many more folks like you may find it useful in the coming future as Xserver dies a slow death.

**Comment by j-rivero** [comment]
> Yeah, the Ubuntu 16.04 use Mir as default display server

Sorry for arriving late to the thread. I'm interesting in this affirmation, could you please point to the information source that explain that 16.04 is using Mir by default?

AFAIK, it won't be the default manager [even for the 16.10 version](http://www.pcworld.com/article/3065294/linux/convergence-delayed-unity-8-wont-be-the-default-desktop-in-ubuntu-1610.html).
